internal/pkg/manifest/validate.go

```diff
@@ -1721,6 +1754,79 @@ func validateDepsForEssentialContainers(deps map[string]containerDependency) err
 	return nil
 }
 
+func validateExposedPorts(opts validateExposedPortsOpts) error {
+	exposedPorts := make(map[uint16]string)
+
+	if opts.mainContainerPort != nil {
+		exposedPorts[aws.Uint16Value(opts.mainContainerPort)] = opts.mainContainerName
+	}
+
+	for name, sidecar := range opts.sidecarConfig {
+		if sidecar.Port == nil {
+			continue
+		}
+		sidecarPort, protocol, err := ParsePortMapping(sidecar.Port)
```

Comment:
huh do we even accept `port/protocol` for sidecars, I thought it was strictly only for `nlb.port`?

It's odd to check for `nlbValidProtocols` for all sidecars because they might not be an nlb in the manifest at all 🤔

Reply:
I think we will keep the implementation as is for now as we might support `udp` and `tcp` protocols for service connect in the future. Do you want to make separate valid protocols list for sidecars? I thought of reusing the existing ones. But happy to make separate list as well.

Reply:
maybe while working on the NLB validation, we can rename it to `validPortProtocols` or something